frontend/src/features/apartment/ApartmentSalesPage.tsx

```diff
@@ -0,0 +1,296 @@
+import React, {useState} from "react";
+
+import {FetchBaseQueryError} from "@reduxjs/toolkit/query";
+import {Button, Checkbox, Dialog, Fieldset, IconAlertCircleFill, LoadingSpinner} from "hds-react";
+import {useParams} from "react-router-dom";
+import {useImmer} from "use-immer";
+import {v4 as uuidv4} from "uuid";
+
+import {
+    useGetApartmentDetailQuery,
+    useGetApartmentMaximumPriceQuery,
+    useSaveApartmentMaximumPriceMutation,
+} from "../../app/services";
+import {FormInputField, Heading, NavigateBackButton, QueryStateHandler, SaveButton} from "../../common/components";
+import OwnershipsList from "../../common/components/OwnershipsList";
+import {IApartmentAddress, IApartmentDetails, IApartmentMaximumPrice, IOwnership} from "../../common/models";
+import {formatAddress, formatDate, formatIndex, formatMoney} from "../../common/utils";
+import MaximumPriceModalContent from "./components/ApartmentMaximumPriceBreakdownModal";
+
+const MaximumPriceModalError = ({error, setIsModalVisible}) => {
+    const nonFieldError = ((error as FetchBaseQueryError)?.data as {message?: string})?.message || "";
+    return (
+        <>
+            <Dialog.Content>
+                <p>Virhe: {(error as FetchBaseQueryError)?.status}</p>
+                <p>{nonFieldError}</p>
+            </Dialog.Content>
+            <Dialog.ActionButtons>
+                <Button
+                    onClick={() => setIsModalVisible(false)}
+                    variant="secondary"
+                    theme="black"
+                >
+                    Sulje
+                </Button>
+            </Dialog.ActionButtons>
+        </>
+    );
+};
+
+const LoadedApartmentSalesPage = ({apartment}: {apartment: IApartmentDetails}) => {
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const hasValidCalculation = apartment.prices.maximum_prices.confirmed?.valid.is_valid;
+    const initialFormData = {
+        announcementDate: null,
+        dealDate:
+            apartment.prices.maximum_prices.confirmed !== null
+                ? apartment.prices.maximum_prices.confirmed.calculation_date
+                : null,
+        dateOfCount:
+            apartment.prices.maximum_prices.confirmed !== null
+                ? apartment.prices.maximum_prices.confirmed.calculation_date
+                : null,
+        price: null,
+        apartment_share_of_housing_company_loans: hasValidCalculation ? apartment.prices.primary_loan_amount : null,
+        mortgageShare: null,
+        isExcludedFromStatistics: false,
+        index: "",
+    };
```

Comment:
Let's use the same names for variables across the whole application

`announcementDate` => `notificationDate` (`notification_date` in backend)
`dealDate` => `apartmentSaleDate` (Also leave this one empty if the confirmed max price calculation is not valid anymore)
`dateOfCount` This doesn't belong in `formData` as it doesn't have a field
`price` => `purchasePrice` (`purchase_price`)
`mortgageShare` this one can be removed, as its never used
`isExcludedFromStatistics` backend has this one as `include_in_statistics`, the opposite of this, so we would either need to change the naming or invert the field when sending data to the API (can be messy) or we change the naming in backend (This might be the preferred route?)
`index` not used, can be removed